api.go

```diff
@@ -77,6 +77,13 @@ type LineMatch struct {
 	LineFragments []LineFragmentMatch
 }
 
+type Symbol struct {
+	Sym        string
+	Kind       string
+	Parent     string
+	ParentKind string
```

Comment:
I asked this in another PR, but is string the best we can do? Additionally, this needs docstrings with examples.

Reply:
I stored this data in the index file as enums, but I will also do this in memory.